doc/languages-frameworks/php.section.md

```diff
@@ -0,0 +1,72 @@
+# PHP
+
+## User Guide
+
+### Using PHP
+
+#### Overview
+
+Several versions of PHP is available on Nix, as well as a bunch of extensions
+and libraries available for each version of PHP.
+
+The attribute `php` refers to the latest version of PHP that has undergone
+sufficient testing in nixpkgs. This may not be the latest and we tend to upgrade
+to the latest after branch-off for a new NixOS release so we get more testing
+before the next release.
+
+We keep releases as long as they are supported for the entirety of the next
+NixOS release life cycle. See [PHP Supported Versions](https://www.php.net/supported-versions.php).
+
+As for packages we have `php.phpPackages` that contains a bunch of attributes where
+some are suitable as extensions (notable example: `php.phpPackages.imagick`). And some
+are more suitable for command line use (notable example: `php.phpPackages.composer`).
+
+We have a special section within `php.phpPackages` called `php.phpPackages.exts` that
+contain certain PHP modules that may not be part of the default PHP derivation
+(example: `php.phpPackages.exts.opcache`).
+
+The `php.phpPackages.exts.*` attributes are official extensions that originates from
+the mainline PHP project while other extensions within the `php.phpPackages.*`
+attribute is a mix of sources (such as `pecl` and other places).
+
+The different versions of PHP that nixpkgs fetch is located under attributes
+named based on major and minor version number (example: `php74` is PHP 7.4).
+
+The same pattern with the versioning applies to the packages attribute set and
+to the base attributes.
+
+The packages attribute set for PHP 7.4 is named `php74.phpPackages` and the attribute
+for PHP 7.4 base package is named `php74base`.
+
+#### Installing PHP with packages
+
+There's two different kinds of things you could install:
+ - A command line utility. Simply refer to it via `php*.phpPackages.*`, and it
+   automatically comes with the necessary PHP environment, certain extensions
+   and libraries around it.
+ - A PHP interpreter with certain extensions available.
+   The `php` attribute provides `php.buildEnv` that allows you to wrap the PHP
+   derivation with an additional config file that makes PHP import additional
+   libraries or dependencies.
+
+##### Example setup for `phpfpm`
+
+Example to build a PHP with `imagick` and `opcache` enabled, and configure it for
+the "foo" `phpfpm` pool:
+
+```nix
+let
+  myPhp = php.buildEnv { exts = pp: with pp; [ imagick exts.opcache ]; };
```

Comment:
Would it make sense to mirror other language infrastructures like Python, which use `withPackages`?

There is also https://discourse.nixos.org/t/wip-rfc-a-new-nixpkgs-frontend-for-language-infrastructures/3447

Reply:
It would, `php.buildEnv` does more things than just adding packages though. And it's something that already exists in `master`. So just renaming it wouldn't make sense. But we can probably add a wrapper for `buildEnv` named `withPackages`. But it's not a super big thing and could be done later on as well.

Reply:
I thought `withPackages` was explicitly rejected by someone... (@FRidh maybe) because that would be misleading. I'm having trouble revealing the details at the moment.

Reply:
My remark was that `withPackages` should only take a lambda, not an attribute set
https://github.com/NixOS/nixpkgs/pull/79377#issuecomment-583104499

Reply:
@FRidh ah, thanks for clarifying!

Reply:
I think a `withPackages` wrapper can be added later on after this PR as well. It's a quite small thing to add to make it simpler to use. But for now I'm fairly happy with all the progress we have here.

We have a bunch of small things we would like to improve after this PR. But we had enough changes already and tried to stick to "a minimal" amount of changes.

We (me and @talyz) were discussing how we possibly could break it down into several PRs and came to the conclusion that we can't.